### DISCUSSION TITLE
How to run the analyzer with models in custom folder and not .cache

### DISCUSSION THREAD

**Discussion by fadilparves** [deepdoctection/deepdoctection#265] [Q&A]
Hi everyone,

I have moved all the models and config from the .cache folder to other custom folder, how do I make the analyzer work. For example below is how I define the config_overwrite

`config_overwrite = [f"PT.LAYOUT.WEIGHTS_TS={base}/dd_cache/weights/layout/d2_model_0829999_layout_inf_only.ts",
 f"PT.ITEM.WEIGHTS_TS={base}/dd_cache/weights/item/d2_model_1639999_item_inf_only.ts",
 f"PT.CELL.WEIGHTS_TS={base}/dd_cache/weights/cell/d2_model_1849999_cell_inf_only.ts",
 f"OCR.CONFIG.TESSERACT={base}/dd_cache/configs/dd/conf_tesseract.yaml"]`

And I am just using the default model that were downloaded by dd during the first run it just that I put them in different folder. But when I run analyzer = dd.get_dd_analyzer(config_overwrite=config_overwrite)

I get the error below

```---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[80], line 1
----> 1 analyzer = dd.get_dd_analyzer(config_overwrite=config_overwrite)

File ~/anaconda3/envs/civil-research/lib/python3.9/site-packages/deepdoctection/analyzer/dd.py:397, in get_dd_analyzer(reset_config_file, config_overwrite)
    394 if tensorpack_available():
    395     disable_tp_layer_logging()
--> 397 return build_analyzer(cfg)

File ~/anaconda3/envs/civil-research/lib/python3.9/site-packages/deepdoctection/analyzer/dd.py:241, in build_analyzer(cfg)
    238 pipe_component_list: List[PipelineComponent] = []
    240 if cfg.USE_LAYOUT:
--> 241     d_layout = _build_detector(cfg, "LAYOUT")
    242     layout = _build_service(d_layout, cfg, "LAYOUT")
    243     pipe_component_list.append(layout)

File ~/anaconda3/envs/civil-research/lib/python3.9/site-packages/deepdoctection/analyzer/dd.py:142, in _build_detector(cfg, mode)
    140 profile = ModelCatalog.get_profile(weights)
    141 categories = profile.categories
--> 142 assert categories is not None
    143 if profile.model_wrapper in ("TPFrcnnDetector",):
    144     return TPFrcnnDetector(config_path, weights_path, categories, filter_categories=filter_categories)

AssertionError: 
```

Appreciate if anyone knows how to make this work. Thank you

**Comment by JaMe76** [comment]
There is an environment variable `DEEPDOCTECTION_CACHE` where you can specify an absolute path to a directory. 

I personally haven't tried if that works seamlessly. I should. But let me know, if there is anything that is behind your expectation.

Note however, that you can only specify the base directory. The sub-directories for weights, configs and datasets are fixed for now. 


As a side note and because I have seen this before: The weights you specify must not be a path, but a registered model, i.e. it must be one of the model in

```
import deepdoctection as dd

print(dd.ModelCatalog.CATALOG.keys())
```
